src/Database.hs

```diff
@@ -13,29 +15,38 @@ import Database.PostgreSQL.Simple.SqlQQ
 import Database.PostgreSQL.Simple.Types
 import Types
 
-connectApp :: DatabaseConfig -> IO Connection
-connectApp (DBConfig host db user password dir) = connectPostgreSQL $ BS8.pack $ unpack url
+connectApp :: MonadIO m => DatabaseConfig -> m Connection
```

Comment:
Here's an example of lifting something from IO to MonadIO.  It is not a huge change, but it does make it very slightly easier to work with if you are already working with monad transformers or mtl style.